ricecooker/chefs.py

```diff
@@ -33,36 +31,39 @@
 
 
 
-# SUSHI CHEF BASE CLASS (and backward compatibiliry)
+# SUSHI CHEF BASE CLASS
 ################################################################################
 
-class BaseChef(object):
+class SushiChef(object):
     """
-    The base class that parses command line arguments for sushichef scripts.
-    Sushi chef sctipts call the `main` method as the entry point, which in turn
-    calls the `run` method to performs all the work (see `uploadchannel`).
-
-    This class also provides backward compaibility with old chef scripts.
-    When invoking the a sushi chef script using the old API: \
-        python -m ricecooker uploadchannel chef_script.py --token=123 ...
-    an instance of this class with `compatibility_mode = True` will be created.
-    Calling `BaseChef.run` will call the function `construct_channel` in `chef_module`.
+    This is the base class that all content integration scripts should subclass.
+    Sushi chef scripts call the `main` method as the entry point, which in turn
+    calls the `run` method to do the work (see `uploadchannel` in `commands.py`).
     """
+    CHEF_RUN_DATA = config.CHEF_DATA_DEFAULT  # loaded from chefdata/chef_data.json
+    TREES_DATA_DIR = config.TREES_DATA_DIR    # tree archives and JsonTreeChef inputs
+
 
-    def __init__(self, *args, compatibility_mode=False, **kwargs):
+    def __init__(self, *args, **kwargs):
         """
-        Setup argparse arguments.
+        The SushiChef initialization concerns maintly parsing command line args.
+        Overrride this method in your sushi chef class to add custom arguments.
         """
-        self.compatibility_mode = compatibility_mode
 
-        # argparse setup
+        # ARGPARSE SETUP
+        # We don't want to add argparse help if subclass has an __init__ method
+        subclasses = self.__class__.__mro__[:-2]     # all subclasses after this
+        if any(['__init__' in c.__dict__.keys() for c in subclasses]):
+            add_parser_help = False    # assume subclass' __init__ will add help
+        else:
+            add_parser_help = True
         parser = argparse.ArgumentParser(
-            description="Ricecooker puts your content in the conten server.",
-            add_help=(self.__class__ == BaseChef)  # only add help if not subclassed
+            description="Chef script for uploading content to Kolibri Studio.",
+            add_help=add_parser_help,
```

Comment:
In case anyone is wondering, using `add_help=False` is necessary in cases where a subclass implements its own custom arg parser. This feature is not used much by suchichef classes (I think only MIT blossoms chef and some of Alejando's chefs use it), but used by the  LineCook class (CSV parsing logic).

For passing custom keyword args to chefs, the approach used by @WenyuZhang1992 is much nicer (no need to overload `__init__`):
https://github.com/learningequality/sushi-chef-refugee-response-crisis-advice/blob/master/sushichef.py#L75-L102